libs/eo/transfers/src/lib/eo-transfers-modal.component.ts

```diff
@@ -0,0 +1,161 @@
+    NgIf,
+  ],
+  standalone: true,
+  styles: [
+    `
+      watt-form-field {
+        margin-top: var(--watt-space-l);
+        margin-bottom: var(--watt-space-m);
+      }
+    `,
+  ],
+  template: `<watt-modal
+    #modal
+    [title]="title"
+    size="small"
+    closeLabel="Close modal"
+    [loading]="requestLoading"
+  >
+    <form [formGroup]="form">
+      <watt-form-field>
+        <watt-label>Receiver</watt-label>
+        <input
+          wattInput
+          required="true"
+          type="text"
+          formControlName="tin"
+          [maxlength]="8"
+          data-testid="new-agreement-receiver-input"
+        />
+        <watt-error *ngIf="form.controls.tin.errors"
+          >An 8-digit TIN/CVR number is required</watt-error
+        >
+      </watt-form-field>
+      <watt-form-field>
+        <watt-label>Period</watt-label>
+        <watt-datepicker
+          required="true"
+          formControlName="dateRange"
+          [range]="true"
+          data-testid="new-agreement-daterange-input"
+        />
+        <watt-error *ngIf="form.controls.dateRange.errors"> A date range is required </watt-error>
+      </watt-form-field>
+    </form>
+    <watt-modal-actions>
+      <watt-button
+        variant="secondary"
+        data-testid="close-new-agreement-button"
+        (click)="modal.close(false)"
+        >Close</watt-button
+      >
+      <watt-button
+        data-testid="create-new-agreement-button"
+        [disabled]="!form.valid"
+        (click)="createAgreement()"
+        >Create transfer agreement</watt-button
+      >
+    </watt-modal-actions>
+  </watt-modal>`,
+})
+export class EoTransfersModalComponent {
+  @ViewChild(WattModalComponent) modal!: WattModalComponent;
+  @Input() title = '';
+
+  form = new FormGroup({
+    tin: new FormControl('', [Validators.minLength(8), Validators.pattern('^[0-9]*$')]),
+    dateRange: new FormControl<WattRange>({ start: '', end: '' }, [WattRangeValidators.required()]),
+  });
+  requestLoading = false;
+
+  constructor(
+    private service: EoTransfersService,
+    private store: EoTransfersStore,
+    private cd: ChangeDetectorRef
+  ) {}
+
+  open() {
+    this.modal.open();
+  }
+
+  createAgreement() {
+    if (!this.form.valid || !this.form.controls['dateRange'].value) return;
+
+    const transfer = {
+      startDate: Math.round(new Date(this.form.controls['dateRange'].value.start).getTime() / 1000),
+      endDate: Math.round(new Date(this.form.controls['dateRange'].value.end).getTime() / 1000),
+      receiverTin: this.form.controls['tin'].value as string,
+    };
+
+    this.requestLoading = true;
+    this.service.createAgreement(transfer).subscribe({
+      next: (transfer) => {
+        this.store.addTransfer(transfer);
+        this.requestLoading = false;
+        this.resetFormValues();
+        this.cd.detectChanges();
+        this.modal.close(true);
+      },
+      error: () => {
+        this.requestLoading = false;
```

Comment:
Question: should error handling be part of this PR, or are we doing a follow-up PR for that?

Reply:
We haven't had a look at error handling yet, to be honest, so it's never been a priority :) Mainly because this is a beta product that does not have any users yet.

Maybe you and I should sit down and talk about scenarios? :)